src/main/java/org/ow2/proactive/catalog/report/TableCatalogObjectsDependenciesBuilder.java

```diff
@@ -0,0 +1,283 @@
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.catalog.report;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.PDPage;
+import org.apache.pdfbox.pdmodel.font.PDType0Font;
+import org.jgrapht.Graph;
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.connectivity.ConnectivityInspector;
+import org.jgrapht.alg.cycle.CycleDetector;
+import org.jgrapht.alg.cycle.Cycles;
+import org.jgrapht.alg.shortestpath.AllDirectedPaths;
+import org.jgrapht.graph.AsSubgraph;
+import org.jgrapht.graph.DefaultEdge;
+import org.jgrapht.traverse.BreadthFirstIterator;
+import org.ow2.proactive.catalog.callgraph.CallGraphHolder;
+import org.ow2.proactive.catalog.callgraph.GraphNode;
+import org.ow2.proactive.catalog.dto.CatalogObjectDependencies;
+import org.ow2.proactive.catalog.dto.CatalogObjectMetadata;
+import org.ow2.proactive.catalog.service.CatalogObjectService;
+import org.ow2.proactive.catalog.util.SeparatorUtility;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import be.quodlibet.boxable.BaseTable;
+import be.quodlibet.boxable.Cell;
+import be.quodlibet.boxable.Row;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 2019-06-14
+ */
+
+@Component
+public class TableCatalogObjectsDependenciesBuilder {
+
+    @Autowired
+    private CellFactory cellFactory;
+
+    private static final String FONT_NAME = "arial-unicode-ms.ttf";
+
+    private static final String MARGIN = "                  ";
+
+    private static final float CELL_WIDTH = 100f;
+
+    @Autowired
+    private SeparatorUtility separatorUtility;
+
+    /**
+     * This method builds a table of catalog objects dependencies ordered and grouped by bucket and object name.
+     *  In case the oder of the callGraphHolder is zero, an appropriate message is displayed.
+     * @param doc
+     * @param callGraphHolder
+     * @param catalogObjectMetadataList
+     * @param catalogObjectService
+     * @param table
+     * @throws IOException
+     */
+    public void buildCatalogObjectsDependenciesTable(PDDocument doc, CallGraphHolder callGraphHolder,
```

Comment:
this method is quite long and dense. Maybe, you could find a way how to factorize it a bit? but I does not know how so i am not insisting.